### ISSUE TITLE
[Good case] Handle errors in defer

### ISSUE THREAD

**Issue by kulti** [firefart/nonamedreturns#5]
I hate named returns too :)

Well, one case are good and should be allowed by linter I suppose.

``` go
func f() (retErr error) {
	err := errors.New("foo")
	defer func() {
		if retErr == nil { // <-- analyse and replace returning error
			retErr = errors.New("bar")
		}
	}()
	return err
}
```

**Comment by firefart** [comment]
will leave this issue open until dependabot has picked it up on the `golangci-lint` repo

**Comment by firefart** [comment]
TODO once v1.0.3 is merged into golangci-lint (job runs sunday):

- [ ] add new option to golangci-lint
- [ ] update tests in golangci-lint

Reference: https://github.com/golangci/golangci-lint/pull/2915

**Comment by maratori** [comment]
Option is already added to golangci-lint (master branch)

**Comment by maratori** [comment]
golangci/golangci-lint#2944
